starlite/datastructures/cookie.py

```diff
@@ -51,3 +51,18 @@ def to_header(self, **kwargs: Any) -> str:
             if value is not None:
                 simple_cookie[self.key][key] = value
         return simple_cookie.output(**kwargs).strip()
+
+    def __eq__(self, other: Any) -> bool:
```

Comment:
I'm actually not sure if this is a good approach. It doesn't really do a strict equality check, since it does not compare for value and other attributes. Maybe this shouldn't be a dunder method, but a separate `is_equal` method with a `strict` flag defaulting to `True`. We could still implement `__eq__` as a callback to  `self.is_equal(strict=True)` then.

I'd also do the canonical "return `NotImplemented`" if the types don't match

Reply:
cookie equality is based on these parameters in browsers, thats the reason. Why wouldnt this be a good idea in your opinion?

Reply:
It makes sense in the HTTP world, but less so in the python world. In the python world it might be expected that two cookies are only equal if their public attributes all equal.

You have a point though, for this specific implementation it makes sense to check equality like this. Probably should be mentioned in the docs somewhere then.